datacube/scripts/ingest.py

```diff
@@ -408,19 +409,30 @@ def ingest_cmd(index,
 
     if config_file:
         config = load_config_from_file(config_file)
+    elif load_tasks:
+        config, tasks = load_tasks_(load_tasks)
+    else:
+        click.echo('Must specify exactly one of --config-file, --load-tasks')
+        sys.exit(-1)
+
+    try:
+        # ignore the added filename key which is not part of the schema
+        tmp_config = {key:config[key] for key in config if key != 'filename'}
```

Comment:
Is this for when going via `load_tasks_(...)`? Can we just drop it from `config`, is it being used later on? Or maybe add it into schema as an optional string parameter?

Reply:
I'm hesitant to add filename to the schema because it is created and used internally by the ingestor. I think that the schema should only contain fields that a user writing an ingestor configuration file would be concerned about. That allows the schema to also be used as documentation. Instead of copying the dictionary which is O(n) complexity I have changed to temporarily deleting the filename key for validation, delete and set are O(1) operations.